### ISSUE TITLE
Resource: trigger

### ISSUE THREAD

**Issue by kvrhdn** [kvrhdn/terraform-provider-honeycombio#2]
A resource to create triggers.

More information: https://docs.honeycomb.io/working-with-your-data/triggers/
API: https://docs.honeycomb.io/api/triggers/

Possible Terraform config (this is not final + probably not legal Terraform code):

```hcl
resource "honeycombio_trigger" "trigger" {
  name        = "Requests are slower than usual"
  description = "Average duration of all requests for the last 5 minutes."

  # NOTE: maybe `enabled` would be simpler to grok?
  disabled = false

  # NOTE: should we prefer using terminology from the UI? I.e. `visualize`, `where`, `group by`
  query {
    # exactly one calculation is required
    calculation {
      op     = "AVG"
      column = "duration_ms"
    }

    # zero or more filter blocks
    filter {
      column = "trace.parent_id"
      op     = "does-not-exist"
    }

    # should also be supported: filter_combination, breakdowns
  }

  frequency = 300 // in seconds, 5 minutes

  threshold {
    op    = ">"
    value = 1000
  }

  # zero or more recipients
  recipient {
    type   = "email"
    target = "hello@example.com"
  }

  recipient {
    type = "pagerduty"
  }
}
```

**Comment by kvrhdn** [comment]
Trigger API is now implemented in `go-honeycombio` since [v0.0.5](https://github.com/kvrhdn/go-honeycombio/releases/tag/v0.0.5).

**Comment by fitzoh** [comment]
I like the  DSL but I might consider separating it from the resource.

I'm thinking of something along the lines of how AWS IAM policies are managed with the [aws_iam_policy_document data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document).

```hcl
data "honeycombio_query" "my_query" {
  calculation {
    ...
  }
  filter {
    ...
  }
}
resource "honeycombio_trigger" "my_trigger" {
    query = data.honeycombio_query.my_query.rendered
}
```

**Comment by kvrhdn** [comment]
Oh, thanks for the link! I didn't know that was possible. This would be really nice, especially since this would allow us to share `honeycombio_query` between triggers and boards.

**Comment by kvrhdn** [comment]
This is now implemented with #10.

Example code: https://github.com/kvrhdn/terraform-provider-honeycombio/blob/main/example/trigger/main.tf
